### ISSUE TITLE
Duplicates and naming

### ISSUE THREAD

**Issue by Evangelink** [fsprojects/FSharpx.Extras#325]
Hi,

First of all, thank you for this helpful library.

I'd like to report here a duplication of the same method with 3 names (ofBoolAndValue, fromTryPattern, tryParseWith).
let a = Double.TryParse "2" |> Option.ofBoolAndValue
let b = Option.fromTryPattern Double.TryParse "2"
let c = Option.tryParseWith Double.TryParse "2"

Also, I was wondering if you should not rename the Option.concat to Option.flatten or Option.switch (to look like Observable.switch) because the name is quite misleading and actually doesn't concat some options.

Cheers

**Comment by mausch** [comment]
Good point about those duplicate functions. Too many people contributing without discussing things I guess.

About Option.concat, it's the same concept as the built-in [`List.concat`](https://msdn.microsoft.com/en-us/library/ee370370.aspx) i.e. `m m a -> m a`

**Comment by Evangelink** [comment]
Don't know how you want to handle things but if you need some help, feel free to ask/contact me.

**Comment by sideeffffect** [comment]
> `m (m a) -> m a`

shouldn't we name it [`join`](https://hackage.haskell.org/package/base/docs/Control-Monad.html#v:join)

**Comment by gdziadkiewicz** [comment]
4th variant: 
```f#
let d = Double.TryParse "2" |> toOption
```
https://github.com/fsprojects/FSharpx.Extras/blob/master/src/FSharpx.Extras/Prelude.fs#L89-L92
